### PR TITLE
Refactors components and adds responsive container

### PR DESCRIPTION
**Pull request by j-zhao**
Refactors structure of how home page is rendered with helper function to
generate responsive wrapper.

Not sure how I feel about how I structured it, but at the very least I think the independent components are clearly separated, and reusable logic is independent from the actual structure of the app.

Idk JS/react.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,116 +1,31 @@
...
 import { HashRouter as Router, Route, Link } from 'react-router-dom';
+import Markdown from './MarkdownItRender.js';
+import ResponsiveContainer from './ResponsiveContainer.js';
 import './App.scss';
+import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
 
-const history = createBrowserHistory();
+function App() {
+  return (
+    <ResponsiveContainer childElement={HomeRouter()} />
```
Comment (shadanan) [inline src/components/App.js]: Can we put the HomeRouter inline here?

File: src/components/MarkdownItRender.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { createBrowserHistory } from 'history';
+import MarkdownIt from 'markdown-it';
+import MarkdownItMusic from 'markdown-it-music'
+import queryString from 'query-string'
+
+const history = createBrowserHistory();
+
+class MarkdownMusic extends React.Component {
```
Comment (shadanan) [inline src/components/MarkdownItRender.js]: Let's move this to a new file.

File: src/components/MarkdownItRender.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  render() {
+    this.md.setTranspose(this.props.transpose);
+    return (
+      <span dangerouslySetInnerHTML={{ __html: this.md.render(this.props.source) }} />
+    );
+  }
+}
+
+class Markdown extends React.Component {
```
Comment (shadanan) [inline src/components/MarkdownItRender.js]: We need a better name for this component. I'm not sure what it is, but here's what this component does:

- Retrieves the markdown file from a GitHub repo
- Provides control of the transpose
- Decodes the Markdown and renders it with MarkdownMusic

Right now, I think that this component is probably trying to do too many things. Let's make a TODO to decouple retrieval of source markdown and controlling arguments to MarkdownMusic.

File: src/components/App.js (hunk 1 above)
Comment (j-zhao) [reply src/components/App.js]: I was conflicted. My ultimate reasoning to breaking out is in the future, when we have potentially future routes, that will be defined in `HomeRouter`, whereas `App` should only contain UI elements and no logic.  Let me know if you disagree.

File: src/components/MarkdownItRender.js (hunk 3 above)
Comment (j-zhao) [reply src/components/MarkdownItRender.js]: Agree, I looked at this for a little bit and it was totally initially vague and unintuitive to me.

**Pull request closed without merging**